File: java/factory/InternationalPaymentMethod.java
package factory;

public abstract class InternationalPaymentMethod implements PaymentMethod {

    private String cardHolder;
    private String cardNumber;
    private String cvv;
    private String expiryDate;

    public InternationalPaymentMethod(String cardHolder, String cardNumber, String cvv, String expiryDate) {
        this.cardHolder = cardHolder;
        this.cardNumber = cardNumber;
        this.cvv = cvv;
        this.expiryDate = expiryDate;
    }

    public abstract void authorizePayment();
    public abstract void startMoneyTransfer();
    public abstract void calculatePaymentFees(double amount);

    public String getCardHolder() {
        return cardHolder;
    }

    public void setCardHolder(String cardHolder) {
        this.cardHolder = cardHolder;
    }

    public String getCardNumber() {
        return cardNumber;
    }

    public void setCardNumber(String cardNumber) {
        this.cardNumber = cardNumber;
    }

    public String getCvv() {
        return cvv;
    }

    public void setCvv(String cvv) {
        this.cvv = cvv;
    }

    public String getExpiryDate() {
        return expiryDate;
    }

    public void setExpiryDate(String expiryDate) {
        this.expiryDate = expiryDate;
    }
}

File: typescript/patterns/factory/international-payment-method.ts
import { PaymentMethod } from "./payment-method-factory";

export abstract class InternationalPaymentMethod implements PaymentMethod {
    private cardHolder: string;
    private cardNumber: string;
    private cvv: string;
    private expiryDate: string;
  
    constructor(
      cardHolder: string,
      cardNumber: string,
      cvv: string,
      expiryDate: string
    ) {
      this.cardHolder = cardHolder;
      this.cardNumber = cardNumber;
      this.cvv = cvv;
      this.expiryDate = expiryDate;
    }
  
    abstract authorizePayment(): void;
    abstract startMoneyTransfer(): void;
    abstract calculatePaymentFees(amount: number): void;
  
    getCardHolder(): string {
      return this.cardHolder;
    }
  
    setCardHolder(cardHolder: string): void {
      this.cardHolder = cardHolder;
    }
  
    getCardNumber(): string {
      return this.cardNumber;
    }
  
    setCardNumber(cardNumber: string): void {
      this.cardNumber = cardNumber;
    }
  
    getCvv(): string {
      return this.cvv;
    }
  
    setCvv(cvv: string): void {
      this.cvv = cvv;
    }
  
    getExpiryDate(): string {
      return this.expiryDate;
    }
  
    setExpiryDate(expiryDate: string): void {
      this.expiryDate = expiryDate;
    }
  }
  